### ISSUE TITLE
Naming/ConstantName flags conditional assignments of classes to camel-cased constants

### ISSUE THREAD

**Issue by patbl** [rubocop-hq/rubocop#6100]
## Expected behavior

I expected this not to trigger the `Naming/ConstantName` cop:

```
class Foo; end
class Bar; end

Lol = if true then Foo else Bar end
```

The docs [say](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/ConstantName)

> To avoid false positives, it ignores cases in which we cannot know for certain the type of value that would be assigned to a constant.

The changelog [says](https://github.com/rubocop-hq/rubocop/blob/5ea73b3453a741b3ec05f781459741f39400c1a8/CHANGELOG.md#0540-2018-03-21)

> Fix false positive in `Naming/ConstantName` when using conditional assignment. (@drenmi)

## Actual behavior

That code triggered the `Naming/ConstantName` cop.

## Steps to reproduce the problem

```
~  ➜ cat foo.rb
class Foo; end
class Bar; end

Lol = if true then Foo else Bar end
~  ➜ rubocop --only Naming/ConstantName foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:4:1: C: Naming/ConstantName: Use SCREAMING_SNAKE_CASE for constants.
Lol = if true then Foo else Bar end
^^^

1 file inspected, 1 offense detected
```

This isn't a problem without the conditional:

```
~  ➜ cat bar.rb
class Foo; end
Bar = Foo
~  ➜ rubocop --only Naming/ConstantName bar.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```
~  ➜ rubocop -V
0.58.1 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-darwin17)
```
